### PR TITLE
[Fix]Flakey tests on LLC and SwiftUI

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Fix flakiness on LLC and SwiftUI tests.

#### LLC
The main reasons for our failing tests seems to be:
1. Updating cancellables storage from multiple threads
2. Using assert methods that are simply waiting for a specified amount of time (i didn't eliminate all of them as there are many, i just fixed the ones failing very often)
3. RTCVideoTracks need to be set to `enabled = false` before deallocation

#### SwiftUI
The main reasons for our failing tests seems to be:
1. Our AudioSession wrapper needs to disable the RTCAudioSession on deinit
2. PeerConnectionFactory needs to cleanUpSSL on deinit (thanks WebRTC for the [docs](https://github.com/ahoyconference/webrtc-ios/blob/master/WebRTC.framework/Headers/RTCSSLAdapter.h). I also did some cleanup for the factory as it seems it had many unnecessary things on it (e.g being an actor when it doesn't maintain any local state, using static methods)
3. Lazily create AudioSession objects
4. Small rewrites on PiPAdapter
5. Make sure that RTCVideoTracks created from Tests are being disabled on tearDown

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: StreamVideoTests/StreamVideo_Tests.swift [hunk 1]
```diff
@@ -143,10 +143,6 @@ final class StreamVideo_Tests: StreamVideoTestCase {
             streamVideo.state.activeCall == nil
                 && streamVideo.state.ringingCall == nil
         }
-
-        // Then
-        XCTAssert(streamVideo.state.ringingCall == nil)
```
Comment (martinmitrevski) [inline StreamVideoTests/StreamVideo_Tests.swift:148]: Why is this removed?

File: Sources/StreamVideo/Utils/AudioRecorder/StreamCallAudioRecorder.swift [hunk 2]
```diff
@@ -102,11 +105,30 @@ open class StreamCallAudioRecorder: @unchecked Sendable {
     open func startRecording(ignoreActiveCall: Bool = false) async {
         do {
             let audioRecorder = try await setUpAudioCaptureIfRequired()
-            guard hasActiveCall || ignoreActiveCall, !isRecording else {
-                log
-                    .info(
-                        "🎙️Attempted to start recording but failed. hasActiveCall:\(hasActiveCall) isRecording:\(isRecording)"
-                    )
+            let startRecordingRequest = StartRecordingRequest(
```
[the remark this answers is not in the document]
Comment (ipavlidakis) [reply Sources/StreamVideo/Utils/AudioRecorder/StreamCallAudioRecorder.swift]: That's right, I moved it here from a cherry pick. Even though it doesn't add change much i'm going to remove it to reduce this PR's scope. Good catch

File: StreamVideoTests/WebRTC/WebRTCClient_Tests.swift [hunk 3]
```diff
@@ -750,10 +784,10 @@ final class WebRTCClient_Tests: StreamVideoTestCase {
         ownCapabilities: [OwnCapability] = [],
         httpClient: HTTPClient? = nil
     ) -> WebRTCClient {
-        let time = VirtualTime()
-        VirtualTimeTimer.time = time
+//        let time = VirtualTime()
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/WebRTCClient_Tests.swift:787]: can we remove the commented out code?

File: StreamVideoTests/StreamVideo_Tests.swift (hunk 1 above)
Comment (ipavlidakis) [reply StreamVideoTests/StreamVideo_Tests.swift:148]: I have removed both lines below
```swift
XCTAssert(streamVideo.state.ringingCall == nil)
XCTAssert(streamVideo.state.activeCall == nil)
```

because the `fulfilment` implementation above, it performs an XCTAssertTrue for the block passed to it which in our case is
```swift
streamVideo.state.activeCall == nil && streamVideo.state.ringingCall == nil
``` and thus verifies the same thing as the remove XCTAsserts.

**Review by martinmitrevski — APPROVED** [review]
